test/fsim/test_main.py

```diff
@@ -79,12 +79,11 @@ def sender(args: argparse.Namespace, mock_queue) -> Sender:
 @pytest.fixture
 async def sender_run(sender: Sender) -> AsyncGenerator[asyncio.Task, None]:
     """Run the `sender` fixture and also send descriptors, and cancel it at the end of the test."""
-    await sender.stream.async_send_heap(sender.descriptor_heap, rate=0.0)
-    task = asyncio.create_task(sender.run(0, False))
-    yield task
-    task.cancel()
-    with pytest.raises(asyncio.CancelledError):
-        await task
+    async with asyncio.TaskGroup() as tg:
+        await sender.stream.async_send_heap(sender.descriptor_heap, rate=0.0)
+        task = tg.create_task(sender.run(0, False))
+        yield task
+        task.cancel()
```

Comment:
Seeing as you removed the `with pytest.raises`, does that mean we don't need to explicitly acknowledge(/catch?) the `asyncio.CancelledError`?
* I gather we're relying on the `TaskGroup` to step in somehow
* At the moment I'm just seeing them as two different entities (`pytest.raises` has a different job to `asyncio.TaskGroup`)

Reply:
TaskGroup's context manager swallows `CancelledError`. Technically I've changed the behaviour since I'm not longer verifying that a CancelledError is raised, but I think I was just abusing `pytest.raises` in the first place as a way to suppress the exception from bubbling out rather than to actually verify that it happened.

Reply:
Noted and noted.